independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/jaxrs/HttpHeadersImpl.java

```diff
@@ -50,10 +49,7 @@ public List<String> getRequestHeader(String name) {
 
     @Override
     public Map<String, Cookie> getCookies() {
-        if (cookies == null) {
-            cookies = Collections.unmodifiableMap(HeaderUtil.getCookies(requestHeaders));
-        }
-        return cookies;
+        return Collections.unmodifiableMap(HeaderUtil.getCookies(requestHeaders));
```

Comment:
I'm a bit worried about us doing this. I see several places calling this to extract one particular cookie and that means we will build a cookie map several times if someone extracts several cookis. Not sure how it could affect performances (I'm more talking about us calling `HeaderUtil.getCookies()` several times, the wrapping probably doesn't cost that much).

Reply:
That's a valid point, but in practice this is only called if users explicitly need to work with cookies, which isn't very common

Reply:
> I think it's probably something that should have some bake time before we backport it.

Sure, fine with me

Reply:
In the original post in SO, they are using it. I wonder if we could somehow taint the cookies when they are modified and regenerated the map if the existing map is tainted?

Just an idea, it might not be worth the hassle or be practical.